lib/Listener/ShareLinkAccessedListener.php

```diff
@@ -65,18 +66,27 @@ public function handle(Event $event): void {
 
 		// Make sure we have a valid limit
 		$token = $event->getShare()->getToken();
-		$limit = $this->config->getAppValue(Application::APP_ID, 'download_limit_' . $token, -1);
+		try {
+			$shareLimit = $this->mapper->get($token);
+			$limit = $shareLimit->getLimit();
 
-		if ($limit > 0) {
-			$count = $this->config->getAppValue(Application::APP_ID, 'download_count_' . $token, 0) + 1;
-			$this->config->setAppValue(Application::APP_ID, 'download_count_' . $token, $count);
+			// Increment this download event
+			$downloads = $shareLimit->getDownloads() + 1;
```

Comment:
potential concurrency concern, I've raised it https://github.com/nextcloud/files_downloadlimit/issues/8 as it doesn't look that critical and rather unlikely to be triggered by a regular user